src/Types.re

```diff
@@ -7,6 +7,84 @@ type requestPolicy = [
   | [@bs.as "cache-and-network"] `CacheAndNetwork
 ];
 
+/* OperationType for the active operation.
+   Use with operationTypeToJs for proper conversion to strings. */
+[@bs.deriving jsConverter]
+type operationType = [
+  | [@bs.as "query"] `Query
+  | [@bs.as "mutation"] `Mutation
+  | [@bs.as "subscription"] `Subscription
+  | [@bs.as "teardown"] `Teardown
+];
+
+/* Cache outcomes for operations. */
+[@bs.deriving jsConverter]
+type cacheOutcome = [
+  | [@bs.as "miss"] `Miss
+  | [@bs.as "partial"] `Partial
+  | [@bs.as "hit"] `Hit
+];
+
+/* Debug information on operations. */
+type operationDebugMeta = {
+  source: option(string),
+  cacheOutcome: option(cacheOutcome),
+  networkLatency: option(int),
+  startTime: option(int),
+};
+
+/* The operation context object for a request. */
+type operationContext = {
+  additionalTypenames: option(array(string)),
+  fetch: option((string, Fetch.requestInit) => Js.Promise.t(Fetch.response)),
+  fetchOptions: option(Fetch.requestInit),
+  requestPolicy,
+  url: string,
+  pollInterval: option(int),
+  meta: option(operationDebugMeta),
+  suspense: option(bool),
+  preferGetMethod: option(bool),
```

Comment:
Updated `operationContext` record type. Includes `additionalTypenames, `fetch`, `suspense`, and `preferGetMethod`.